packages/@aws-cdk/aws-kinesisfirehose/lib/lambda-function-processor.ts

```diff
@@ -0,0 +1,32 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct } from 'constructs';
+import { DataProcessorBindOptions, DataProcessorConfig, DataProcessorProps, DataProcessorIdentifier, IDataProcessor } from './processor';
+
+/**
+ * Use an AWS Lambda function to transform records.
+ */
+export class LambdaFunctionProcessor implements IDataProcessor {
+  /**
+   * The constructor props of the LambdaFunctionProcessor.
+   */
+  public readonly props: DataProcessorProps;
+
+  private readonly processorIdentifier: DataProcessorIdentifier;
+
+  constructor(private readonly lambdaFunction: lambda.IFunction, props: DataProcessorProps = {}) {
+    this.props = props;
```

Comment:
shorthand?
```suggestion
  constructor(private readonly lambdaFunction: lambda.IFunction, public readonly props: DataProcessorProps = {}) {
```

Reply:
I get a linter error if I don't provide a docstring for `props`. Maybe I am doing something wrong there?

Reply:
that's odd because it should be pulled from IDataProcessor. no biggie tho

Reply:
Believe me, I agree with you it should! I am not sure why it still complains.